src/main/scala/zio/s3/settings.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.s3
+
+import java.net.URI
+
+import software.amazon.awssdk.regions.Region
+import zio.{ IO, ZIO }
+
+import scala.util.Try
+
+final case class S3Credentials(accessKeyId: String, secretAccessKey: String)
+
+final private[s3] case class S3Settings(region: Region, credentials: S3Credentials, uriEndpoint: Option[URI])
```

Comment:
when `S3Settings` is private I cannot write in DIstage:
```
make[S3.Service[Any]].fromResource(zio.s3.S3.Live.connect(_: S3Settings))
```

Reply:
thanks for your early comments. I will try to fix it

Reply:
```
val f: S3Settings => Managed[S3ConnectionFailure, S3] = S3.Live.connect(_: S3Settings)
```

this compile fine, i dont really understand, maybe it is a distage problem, i never use it